### PR TITLE
fix index out of range error when trying to access client-config arra…

### PR DESCRIPTION
**Pull request by RamLavi**
…y in empty webhook instance

in order to avoid accessing client-config (that holds the caBundle in the mutating/validating webhook configuration instance) in a case where such webhook section does not exist, we added a protection check.

Signed-off-by: Ram Lavi <ralavi@redhat.com>

### CODE REVIEW

File: pkg/webhook/server/certificate/configuration.go [hunk 1]
```diff
@@ -109,5 +109,10 @@ func (m *Manager) CABundle() ([]byte, error) {
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get %s webhook configuration %s", m.webhookType, m.webhookName)
 	}
-	return m.clientConfigList(webhook)[0].CABundle, nil
+
+	if clientConfigList := m.clientConfigList(webhook); len(clientConfigList) == 0 {
+		return nil, errors.Wrapf(err, "failed to get clientConfig in %s webhook configuration %s", m.webhookType, m.webhookName)
```
Comment (ormergi) [inline pkg/webhook/server/certificate/configuration.go:114]: Can you add  "CABundle from" to the error message?
`"failed to get CABundle from clientConfig in %s webhook configuration %s"`

File: pkg/webhook/server/certificate/configuration.go [hunk 2]
```diff
@@ -109,5 +109,10 @@ func (m *Manager) CABundle() ([]byte, error) {
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get %s webhook configuration %s", m.webhookType, m.webhookName)
 	}
-	return m.clientConfigList(webhook)[0].CABundle, nil
+
+	if clientConfigList := m.clientConfigList(webhook); len(clientConfigList) == 0 {
```
Comment (qinqon) [inline pkg/webhook/server/certificate/configuration.go:113]: else + returns is a "no-no"

Something like this is more idiomatic with golang and with the rest of the code here.
```golang
clientConfigList := m.clientConfigList(webhook)
if len(clientConfigList) == 0 {
   return nil, errors.Wrapf(err, "failed to get clientConfig in %s webhook configuration %s", m.webhookType, m.webhookName)
}

return clientConfigList[0].CABundle, nil
```

File: pkg/webhook/server/certificate/manager.go [hunk 3]
```diff
@@ -117,7 +117,7 @@ func (m *Manager) Start(stopCh <-chan struct{}) error {
 // of generating new one
 func (m *Manager) loadCACertFromCABundle() {
 	caBundle, err := m.CABundle()
-	if err != nil || len(caBundle) == 0 {
```
Comment (qinqon) [inline pkg/webhook/server/certificate/manager.go:120]: This is not related and also `ParseCertsPEM` will fail if caBundle is empty,

File: pkg/webhook/server/certificate/configuration.go [hunk 4]
```diff
@@ -109,5 +109,10 @@ func (m *Manager) CABundle() ([]byte, error) {
...
+		return nil, errors.Wrapf(err, "failed to get clientConfig in %s webhook configuration %s", m.webhookType, m.webhookName)
+	} else {
+		return clientConfigList[0].CABundle, nil
```
Comment (ormergi) [inline pkg/webhook/server/certificate/configuration.go:116]: Shouldn't we also check if CABundel is not nil?

File: pkg/webhook/server/certificate/configuration.go (hunk 2 above)
Comment (RamLavi) [reply pkg/webhook/server/certificate/configuration.go:113]: ack

File: pkg/webhook/server/certificate/configuration.go (hunk 4 above)
Comment (qinqon) [reply pkg/webhook/server/certificate/configuration.go:116]: Agree let's add that check to first == nil then len 0 with different error message so we can differenciate.

File: pkg/webhook/server/certificate/manager.go [hunk 5]
```diff
@@ -117,7 +117,7 @@ func (m *Manager) Start(stopCh <-chan struct{}) error {
 // of generating new one
```
Comment (ormergi) [inline pkg/webhook/server/certificate/manager.go:117]: Hey what will happen if cert-manager is restarted and current CA is dummy cert
will it trigger new cert creation?
@RamLavi @qinqon

File: pkg/webhook/server/certificate/manager.go [hunk 6]
```diff
@@ -117,7 +117,7 @@ func (m *Manager) Start(stopCh <-chan struct{}) error {
...
 func (m *Manager) loadCACertFromCABundle() {
 	caBundle, err := m.CABundle()
-	if err != nil || len(caBundle) == 0 {
+	if err != nil {
```
Comment (ormergi) [inline pkg/webhook/server/certificate/manager.go:120]: Hey what will happen if cert-manager is restarted and the current CA is dummy or empty cert,
will it trigger new cert creation?
@RamLavi @qinqon

File: pkg/webhook/server/certificate/manager.go (hunk 3 above)
Comment (RamLavi) [reply pkg/webhook/server/certificate/manager.go:120]: If I remove this line entirely then I think we could have an issue in ParseCertsPEM(), because there we ask `for len(caBundle) > 0 {` so we might fail on that

File: pkg/webhook/server/certificate/manager.go (hunk 6 above)
Comment (RamLavi) [reply pkg/webhook/server/certificate/manager.go:120]: I think so yes. is that a problem?
